gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaAvroJobStatusMonitor.java

```diff
@@ -165,6 +177,26 @@ public GobblinTrackingEvent deserializeEvent(DecodeableKafkaRecord<byte[],byte[]
         break;
       case TimingEvent.JOB_COMPLETION_PERCENTAGE:
         properties.put(TimingEvent.JOB_LAST_PROGRESS_EVENT_TIME, properties.getProperty(TimingEvent.METADATA_END_TIME));
+        break;
+      case JobEvent.WORK_UNITS_CREATED:
+        Long numWorkUnits = Long.parseLong(properties.getProperty(CountEventBuilder.COUNT_KEY));
```

Comment:
There is a quite sizable chunk of logic in this "case" branch, you can move it to a separate method to make the current one smaller and easier to reason about.

Reply:
Just noticed that current method is called "parseJobStatus", and looks like we are doing something extra here. You can also consider moving this logic to KafkaJobStatusMonitor.processMessage() , which can be more fitting.